drivers/mm/mm_drv_intel_adsp_mtl_tlb.c

```diff
@@ -667,7 +667,7 @@ static int sys_mm_drv_mm_init(const struct device *dev)
 	    L2_SRAM_BASE > UNUSED_L2_START_ALIGNED) {
 
 		__ASSERT(false,
-			 "unused l2 pointer is outside of l2 sram range %p\n",
+			 "unused l2 pointer is outside of l2 sram range %lu\n",
```

Comment:
Wait, is this not going to change the display from hex to decimal?

Why not just cast to `(void *)` instead?

Reply:
@marc-hb  I think you are right, we want to print the pointer in hex. Fixed this plus commit message.

Thanks!